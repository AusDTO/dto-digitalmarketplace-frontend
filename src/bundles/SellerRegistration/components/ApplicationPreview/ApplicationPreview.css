.center {
  max-width: 75em;
  margin: 0 auto;
  padding: 0 1.6em;
}

.uikit-display-4 {
  font-family: -apple-system, system-ui, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
  font-size: 18px;
  font-weight: 700;
  height: 25px;
  line-height: 25.2px;
  margin-bottom: 7.2px;
  margin-top: 0px;
}