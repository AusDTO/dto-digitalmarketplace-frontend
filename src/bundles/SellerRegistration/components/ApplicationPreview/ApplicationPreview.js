import React from 'react';
import {connect} from 'react-redux';
import { Link } from 'react-router';
import ConnectedLink from '../ConnectedLink';
import isEmpty from 'lodash/isEmpty';

import { Body, ReviewHeader, PrivateInfo } from '../../../../shared/SellerProfile';

import apppreview from './application-preview.css'; // eslint-disable-line no-unused-vars

const ApplicationPreview = ({ header, body, privateInfo, onClick }) => (
  <div>
      {onClick && <div className="row">
      <div className="callout--calendar-event col-sm-8 col-xs-12">
        <h3>Preview your profile</h3>
        <p>Buyers will see your business information previewed below. If the information is correct, continue to the final step to submit your application or continue editing.</p>
        <Link to="/submit">{({ href }) => (
            <a
              href={href}
              role="button"
              onClick={onClick}
            >Save and continue</a>
          )
        }</Link>
          <Link  to="/review">Go back and edit</Link>
      </div>
    </div>}
    <ReviewHeader 
      {...header}
    />
    <div className="row" styleName="apppreview.center">
      <div className="col-sm-8 col-xs-12">
        <Body {...body} />
        {!onClick && privateInfo.disclosures && <PrivateInfo {...privateInfo} />}
      </div>
    </div>
  </div>
);

const mapStateToProps = ({ application }, { documentsUrl, onClick, ...rest }) => {
    let {
      name,
      seller_type,
      summary,
      website,
      contact_email,
      contact_phone,
      contact_name,
      services: provides,
      case_studies,
      travel: interstate,
      number_of_employees,
      local_government_experience,
      state_government_experience,
      federal_government_experience,
      other_panels,
      disclosures,
      documents_url = '../documents/',
      case_study_url,
      ...body
    } = application;

    let caseStudyLink = null;
    if (typeof case_study_url !== 'undefined') {
      caseStudyLink = ({id, children}) => (<a href={`${case_study_url}${id}`}>{children}</a>);
    } else {
      caseStudyLink = ({id, children}) => (
        <ConnectedLink to={`${rest.pathname}/case-study/${id}`}>{children}</ConnectedLink>
      );
    }

    // FIXME this is a workaround, somehow python returns empty case studies as a list.
    // Either there is bad data floating around or something is turning empty objects
    // into an array.
    // Body expects object not array.
    if (Array.isArray(case_studies) && isEmpty(case_studies)) {
      case_studies = {};
    }

    return {
        header: {
          name,
          seller_type,
          summary,
          website,
          contact_email,
          contact_phone,
          contact_name,
          number_of_employees
        },
        body: {
          provides,
          case_studies,
          interstate,
          CaseStudyLink: caseStudyLink,
          documentsUrl: documents_url,
          ...body
        },
        privateInfo: (disclosures ? {
            case_studies,
            number_of_employees,
            local_government_experience,
            state_government_experience,
            federal_government_experience,
            other_panels,
            disclosures
        } : {}),
        onClick
    }
}

export {
    mapStateToProps
}

export default connect(mapStateToProps)(ApplicationPreview);