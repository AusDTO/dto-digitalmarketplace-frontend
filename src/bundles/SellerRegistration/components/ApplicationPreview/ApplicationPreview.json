{
  "basename": "/applicant-preview.html",
  "form_options": {
    "action": "https://httpbin.org/post",
    "csrf_token": "abc"
  },
  "privateInfo": {
    "disclosures" : {
      "conflicts_of_interest": "no",
      "conflicts_of_interest_details": "",
      "insurance_claims": "no",
      "insurance_claims_details": "",
      "investigations": "no",
      "investigations_details": "",
      "legal_proceedings": "no",
      "legal_proceedings_details": "",
      "other_circumstances": "no",
      "other_circumstances_details": "",
      "structual_changes": "no",
      "structual_changes_details": ""
    }
  },
  "application": {
    "representative": "Example person",
    "contact_name": "Example Company",
    "abn": "133",
    "contact_phone": "123",
    "contact_email": "example@gmail.com",
    "summary": "good company",
    "website": "https://www.alpha",
    "linkedin": "https://www.linkedin.com/company/Jalpha",
    "address": {
      "addressLine": "123 fake st",
      "suburb": "Town",
      "state": "nsw",
      "postalCode": "2602"
    },
    "services": {
      "User research": true,
      "Architecture": true
    },
    "pricing": {
      "User research": {
        "minPrice": "1",
        "maxPrice": "2"
      },
      "Architecture": {
        "minPrice": "3",
        "maxPrice": "4"
      }
    },
    "case_studies": {
      "7958aa7d-0f21-2cbd-a9a2-0198c2468cb4": {
        "service": "User research",
        "title": "A user can upload files ",
        "client": "dsfsdf",
        "timeframe": " January 2016 — June 2016",
        "opportunity": "fsdfsd",
        "approach": "sfdsds",
        "outcome": [
          "dsfsd",
          "sdffsd"
        ],
        "projectLinks": ["http://wwww"]
      },
      "bcec400b-bf71-22a6-9a84-d5a89431fb1a": {
        "service": "Architecture",
        "title": "sdfsdf",
        "client": "fdgfd",
        "timeframe": " January 2016 — June 2016",
        "opportunity": "fsd",
        "approach": "sdfsdf",
        "outcome": [
          "dsfsdf",
          "sdfsdf"
        ]
      }
    },
    "products": [
        {
            "name": "Avoka Transact for Government",
            "pricing": "http://www.avoka.com/request-a-demo",
            "summary": "Avoka Transact is a digital business platform used to accelerate customer acquisition and increase business agility in financial services, government, education, field service and other industries. The system serves intelligent applications focused on customer acquisition, onboarding and service, creating a digital alternative to aging PDF or paper forms. ",
            "support": "http://www.avoka.com/contact-us?utm_LSD=ContactUs&utm_OLSD=ContactUs",
            "website": "http://www.avoka.com/avoka-transact"
        }
    ],
    "documents": {
      "financial": {
        "filename": "financial.pdf",
        "expiry": "2017-10-21"
      },
      "liability": {
        "filename": "liability.pdf",
        "expiry": "2017-10-21"
      },
      "workers": {
        "filename": "workers.pdf",
        "expiry": "2017-12-25"
      }
    }
  }
}