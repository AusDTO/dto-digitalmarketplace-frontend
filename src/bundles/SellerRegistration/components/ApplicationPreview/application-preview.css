.center {
  max-width: 75em;
  margin: 0 auto;
}