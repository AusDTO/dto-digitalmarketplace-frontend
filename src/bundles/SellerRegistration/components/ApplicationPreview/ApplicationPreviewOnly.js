import React from 'react';
import {connect} from 'react-redux';
import ApplicationPreview from './ApplicationPreview'

const ApplicationPreviewOnly = ({application}) => (
    <ApplicationPreview application={application}/>
);


const mapStateToProps = (state) => {
    return {
        application: state.application
    }
}

export {
    mapStateToProps
}

export default connect(mapStateToProps)(ApplicationPreviewOnly);