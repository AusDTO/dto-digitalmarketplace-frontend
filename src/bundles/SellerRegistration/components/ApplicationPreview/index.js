import ApplicationPreview from './ApplicationPreview';
export default ApplicationPreview;