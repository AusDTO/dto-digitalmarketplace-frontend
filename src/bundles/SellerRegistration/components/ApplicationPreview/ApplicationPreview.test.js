jest.mock('../../../../shared/Icon/_getIcons');

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
import ApplicationPreview from './ApplicationPreview';
import createStore from '../../redux/create';

describe('<ApplicationPreview />', () => {
  it('should render new applicant text', () => {
    const store = createStore({
      application: {
        services: ['service1'],
        assessed_domains: ['service2'],
        case_studies: {},
        supplier: {}
      }
    });
    const wrapper = mount(
    <Provider store={store}>
        <ApplicationPreview />
      </Provider>
    );

    const body = wrapper.find('ApplicationPreview').prop('body');
    expect(body['unassessed']).toContain('0');
    expect(body['assessed']).toContain('service2');
  });
});

test('ApplicationPreview renders with props', () => {
  const store = createStore({
      application: {
        services: ['service1'],
        assessed_domains: ['service2'],
        case_studies: {},
        supplier: {}
      },
      privateInfo: {disclosures: false}
    });
    const component = renderer.create(
    <Provider store={store}>
        <ApplicationPreview />
      </Provider>
    );
    let tree = component.toJSON();
   expect(tree).toMatchSnapshot()
 });
