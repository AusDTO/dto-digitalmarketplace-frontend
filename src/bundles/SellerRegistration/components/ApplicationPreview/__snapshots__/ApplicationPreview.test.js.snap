// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationPreview renders with props 1`] = `
<div>
  <div />
  <section
    styleName="styles.review-header"
  >
    <div>
      <div
        className="row "
      >
        <div
          className="col-sm-8"
        >
          <h1
            className="au-display-xl"
            styleName="styles.heading"
            tabIndex="-1"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <article
          className="col-xs-12 col-md-8 col-sm-7"
        >
          <div
            className="row"
            styleName="styles.badges-row"
          >
            <div
              styleName="styles.badges"
            />
          </div>
          <div
            styleName="styles.seller-profile-summary"
          >
            <p />
          </div>
          <div
            className="row"
            styleName="styles.meta-row"
          >
            <div
              className="col-xs-12 col-sm-3"
            >
              <h2
                className="au-display-lg"
              >
                Website
              </h2>
            </div>
            <div
              className="col-xs-12 col-sm-8 col-sm-push-1"
            >
              <p>
                <a
                  href={undefined}
                  rel="external"
                  styleName="external-link"
                  target="_blank"
                />
              </p>
            </div>
          </div>
        </article>
      </div>
    </div>
  </section>
  <div
    styleName="styles.seller-profile-container"
  >
    <div
      className="row"
      styleName=""
    >
      <div
        className="col-sm-8 col-xs-12"
      >
        <article
          className="seller-profile"
          styleName=""
        >
          <div
            styleName="seller-profile-content"
          >
            <div
              className="row"
              styleName="styles.row"
            >
              <hr />
              <div
                className="col-sm-3 col-xs-12"
              >
                <h2
                  className="seller-profile__section-title au-display-lg"
                >
                  Categories
                </h2>
              </div>
              <div
                className="col-sm-8 col-sm-push-1 col-xs-12"
              >
                <p>
                  This seller has not yet been approved in any categories.
                </p>
              </div>
            </div>
            <div
              className="row"
              styleName="styles.row"
            >
              <hr />
              <div
                className="col-sm-3 col-xs-12"
              >
                <h2
                  className="seller-profile__section-title au-display-lg"
                >
                  Company Details
                </h2>
              </div>
              <div
                className="col-sm-8 col-sm-push-1 col-xs-12"
              >
                <p
                  styleName="bold-noMargin"
                >
                  Authorised representative
                </p>
                <p>
                  <span />
                  <br />
                </p>
              </div>
            </div>
            <div
              className="row"
              styleName="styles.row"
            >
              <hr />
              <div
                className="col-sm-3 col-xs-12"
              >
                <h2
                  className="seller-profile__section-title au-display-lg"
                >
                  Signed agreement
                </h2>
              </div>
              <div
                className="col-sm-8 col-sm-push-1 col-xs-12"
              />
            </div>
          </div>
        </article>
      </div>
    </div>
  </div>
</div>
`;
